lib/features/auth/presentation/bloc/register_cubit/register_cubit.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:final_project/features/auth/domain/entities/register.dart';
+import 'package:final_project/features/auth/domain/use_case/register_use_case.dart';
+import 'package:final_project/features/auth/presentation/bloc/register_cubit/register_states.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:injectable/injectable.dart';
+
+@injectable
+class RegisterCubit extends Cubit<RegisterStates> {
+  RegisterCubit(this._useCase) : super(const RegisterInitialState());
+
+  final RegisterUseCase _useCase;
+
+  static RegisterCubit get(context) => BlocProvider.of(context);
+
+  Future<void> register({
+    required String name,
+    String? phone,
+    required String email,
+    required String password,
+    required String confirmPassword,
+  }) async {
+    emit(const RegisterLoadingState());
+    final register = Register(
```

Comment:
Why not construct the object in UI and pass it to this method?

So it becomes:

Future<void> register({
    required Register params
  })